### PR TITLE
folds(yaml): Fold at the entire node, not at the body.

### PR DESCRIPTION
**Pull request by wookayin**
The current yaml folds are based on #887 and #928, but I don't think this is a good folding syntax.

Example and Tests
-----------------

Let's take a look at the example which is a slight extension of #887's one:

```
days:
  Monday:
    - test: example
      day: 1
  Tuesday:
    - test: example
      day: 2
  Wednesday:
    - test: example
      day: 3
    - test: example
      day: 3
  Thursday:
    - test: example
      day: 4
  Friday:
    - test: example
      day: 5
```

### Current behavior:

(at foldlevel=0)
```
days:  ⋯    ( 17 lines)
```

(at foldlevel=1)
```
days:
  Monday:  ⋯    ( 16 lines)
```

Oops! you can fold the "body" of the `days` node (`block_mapping_pair`). See [this comment from #887](https://github.com/nvim-treesitter/nvim-treesitter/issues/887#issuecomment-874207744) as well.

(at foldlevel=2)
```
days:
  Monday:
    - test: example  ⋯    ( 2 lines)
  Tuesday:
    - test: example  ⋯    ( 2 lines)
  Wednesday:
    - test: example  ⋯    ( 4 lines)
  Thursday:
    - test: example  ⋯    ( 2 lines)
  Friday:
    - test: example  ⋯    ( 2 lines)
```

Again, this is because only the "body" can be folded. But wouldn't we want to see `Monday: ...`, `Tuesday: ...`, etc. only? Folding a list of two items on `Wednesday` into a single line is also confusing.

### Changed behavior:

(at foldlevel=1)
```
days:
  Monday:  ⋯    ( 3 lines)
  Tuesday:  ⋯    ( 3 lines)
  Wednesday:  ⋯    ( 5 lines)
  Thursday:  ⋯    ( 3 lines)
  Friday:  ⋯    ( 3 lines)
```

The corrected behavior.

(at foldlevel=2)
```
days:
  Monday:
    - test: example  ⋯    ( 2 lines)
  Tuesday:
    - test: example  ⋯    ( 2 lines)
  Wednesday:
    - test: example  ⋯    ( 2 lines)
    - test: example  ⋯    ( 2 lines)
  Thursday:
    - test: example  ⋯    ( 2 lines)
  Friday:
    - test: example  ⋯    ( 2 lines)
```

Note folding of "list of items" on the `Wednesday` node (`block_sequence_item`)

I think this is a more semantically meaningful fold, and also meets @gegoune's request in the original message of #887 (separate folds for some "mapping" nodes).

/cc @gegoune @stsewd @vigoux

### CODE REVIEW

**Comment by wookayin** [comment]
This would be a simple fix and PR and I believe it should be blocked by other pending issues. Can the maintainers review and merge this?

**Comment by clason** [comment]
First step would be to get all CI to pass ;)

**Comment by wookayin** [comment]
> "At least 1 approving review is required by reviewers with write access."

in order for the CI jobs can run without being aborted...?

**Comment by clason** [comment]
No, that's a separate requirement.

**Review by theHamsta — APPROVED** [review]
Looks good to me! With this the foldings are more in line with the other languages.

 kevinhwang91/nvim-ufo can be used by anyone who isn't happy with the folded line losing syntax highlighting

**Comment by wookayin** [comment]
Thank you @theHamsta and @clason for your help and merging this!

**Pull request merged**
